### PR TITLE
WIP: Fix Mask/Transitions for Global Timeline Effects

### PR DESCRIPTION
**Pull request by jonoomph**
Refactor of global timeline effects, to address a regression with global/timeline Mask/Transitions no longer working correctly. This was caused by an optimization that broke the general behavior of the global transitions.

### CODE REVIEW

**Comment by ferdnyc** [comment]
@jonoomph It might be a good idea to make the new `GetFrame` signature a `protected` method (and `friend class` the callers that need access). Right now it's ending up in the Python API, and I don't even want to think about how _that_ works when passing a pointer to a struct as one of the arguments.

**Comment by ferdnyc** [comment]
Huh! Well, apparently it works fine. SWIG, you so crazy.
```python3
>>> import openshot
>>> c = openshot.Clip("/var/tmp/pix_scale.mp4")
>>> c.Open()
>>> st = openshot.TimelineInfoStruct()
>>> st.timeline_frame_number = 27
>>> st.is_top_clip = True
>>> f = c.GetFrame(openshot.Frame(), 1, st)
```
The returned `f` is a perfectly valid `1px` × `1px` (due to the default-constructed `background_frame`) instance of `openshot.Frame`. 

So, I guess it's dealer's choice whether the new GetFrame _should_ be part of the public API, or if it's really meant for `TimelineBase` internal use only.

File: src/Timeline.cpp [hunk 1]
```diff
@@ -477,9 +477,15 @@ std::shared_ptr<Frame> Timeline::GetOrCreateFrame(std::shared_ptr<Frame> backgro
 // Process a new layer of video or audio
 void Timeline::add_layer(std::shared_ptr<Frame> new_frame, Clip* source_clip, int64_t clip_frame_number, int64_t timeline_frame_number, bool is_top_clip, float max_volume)
 {
-	// Get the clip's frame, composited on top of the current timeline frame
+    // Create timeline options (with details about this current frame request)
+    TimelineInfoStruct* options = new TimelineInfoStruct();
+    options->timeline_frame_number = timeline_frame_number;
+    options->is_top_clip = is_top_clip;
+
```
[the remark this answers is not in the document]
Comment (ferdnyc) [reply src/Timeline.cpp:484]: (Er, I guess actually that should be `new_frame->number`, here. It'll be accessible _inside_ `GetFrame()` as `background_frame->number`.

[the remark this answers is not in the document]
Comment (ferdnyc) [reply src/Timeline.cpp:484]: (For that matter, same question about the need to pass `timeline_frame_number` to `Timeline::add_layer()`, vs.  just setting `new_frame->number` on the frame passed in as `new_frame`.)

[the remark this answers is not in the document]
Comment (jonoomph) [reply src/Timeline.cpp]: Good catch! Removing this, and refactoring it out of existence.

**Comment by jonoomph** [comment]
Thanks for the feedback @ferdnyc! Merged!

**Pull request merged**